### PR TITLE
Cleanup XQueryURLRewrite

### PR DESCRIPTION
**Pull request by adamretter**
1. Simplify and make the logic for finding a `controller.xql` more transparent.
2. Lots of general code cleanup

### CODE REVIEW

**Comment by adamretter** [comment]
@dizzzz this is a pre-requisite to me fixing it in the Caching and Locking improvements PR

**Comment by dizzzz** [comment]
is this to be merged in the current 'develop' ? not in your locking PR only?

**Comment by adamretter** [comment]
@dizzzz yes current develop please

File: src/org/exist/http/urlrewrite/XQueryURLRewrite.java [hunk 1]
```diff
@@ -726,166 +728,166 @@ protected String adjustPathForSourceLookup(String basePath, String path) {
...
+
+    /**
+     * Finds a `controller.xql` file within a Collection hierarchy.
+     * Most specific collections are considered first.
+     *
+     * For example, given the collectionUri `/db/apps`
+     * and the pathPomponents `['myapp', 'data']`, the
+     * order or search will be:
+     *
+     * /db/apps/myapp/data/collection.xconf
```
Comment (dizzzz) [inline src/org/exist/http/urlrewrite/XQueryURLRewrite.java]: @adamretter Am I correct that in the original code the search is actually reverse?
```
/db/apps/collection.xconf
/db/apps/myapp/collection.xconf
/db/apps/myapp/data/collection.xconf
``` 

Your approach makes more sense to me in a way, but I am not sure bout the consequences? @wolfgangmm

Comment (adamretter) [reply src/org/exist/http/urlrewrite/XQueryURLRewrite.java]: Nope. They both search in the same direction. As I mentioned, the original code is very hard to understand

Comment (dizzzz) [reply src/org/exist/http/urlrewrite/XQueryURLRewrite.java]: I confirm.... the code was unbelievable ugly.

**Pull request merged**